SignalServiceKit/src/Util/TypingIndicators.swift

```diff
@@ -27,14 +27,66 @@ public protocol TypingIndicators: class {
     // TODO: Use this method.
     @objc
     func areTypingIndicatorsVisible(inThread thread: TSThread, recipientId: String) -> Bool
+
+    @objc
+    func setTypingIndicatorsEnabled(value: Bool)
+
+    @objc
+    func areTypingIndicatorsEnabled() -> Bool
 }
 
 // MARK: -
 
 @objc(OWSTypingIndicatorsImpl)
 public class TypingIndicatorsImpl: NSObject, TypingIndicators {
+
     @objc public static let typingIndicatorStateDidChange = Notification.Name("typingIndicatorStateDidChange")
 
+    private let kDatabaseCollection = "TypingIndicators"
```

Comment:
I know we've done this in multiple places, but it seems weird to have an entire collection (table) which exists just to hold one row with one boolean. In my mind it makes sense to use the existing preferences collection for pretty much all singleton data.

I guess it's less straight forward because SSK doesn't know about OWSPreferences?

Reply:
Right, and I'm not eager to move `OWSPreferences ` to SSK.